### PR TITLE
fix a couply of clippy findings

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

**Comment by ytmimi** [comment]
I think there's some overlap here with what's being proposed in https://github.com/rust-lang/rustfmt/pull/5400

File: src/bin/main.rs [hunk 1]
```diff
@@ -387,16 +387,9 @@ fn format_and_emit_report<T: Write>(session: &mut Session<'_, T>, input: Input)
...
+    matches!(term::stderr(), Some(ref t) if session.config.color().use_colored_tty()
                 && t.supports_color()
-                && t.supports_attr(term::Attr::Bold) =>
-        {
-            true
-        }
-        _ => false,
-    }
+                && t.supports_attr(term::Attr::Bold))
 }
```
Comment (ytmimi) [inline src/bin/main.rs:393]: #5400 is using `is_some_and`, which I think it a little clearer than the `matches!` that you're using here.

```rust
    term::stderr().is_some_and(|t| {
        session.config.color().use_colored_tty()
            && t.supports_color()
            && t.supports_attr(term::Attr::Bold)
    })
```

**Comment by matthiaskrgr** [comment]
looking at the commits of https://github.com/rust-lang/rustfmt/pull/5400 , I skipped most of these warnings because I didn't find them very interesting but yes there might be some overlap besides that.

File: src/bin/main.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/bin/main.rs]: fixed

**Comment by ytmimi** [comment]
Out of curiosity, what about these changes did you find interesting?

**Comment by matthiaskrgr** [comment]
I'm usually trying to go for stuff that imo makes code slightly more elegant, simpler or easier to understand (for me, at least :upside_down_face: ) / doing the same things with less code, ĺike using an array instead of vec when its sufficient, having a nice flow of iterator chains that help building a metal model what is going on etc.

Unnecessary needles borrows which I didnt touch can sometimes help keep in mind that a var is borrowed, even if it is already a reference (although I think rust-analyzer can show you that nowadays..?)

IIRC Caleb once mentioned that he liked the if nesting as is, which is why I didn't touch `collapsible_else_if`  (this is kind of a personal-taste lint, sometimes it makes sense, but not always imo.

**Review by ytmimi — APPROVED** [review]
Thanks for explaining. Overall I think these changes look reasonable!

**Pull request merged**